spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -345,5 +345,26 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   def predictClass(dataset: RDD[Sample[T]]): RDD[Int] = {
     Predictor(this).predictClass(dataset)
   }
+
+  /**
+    * Set weight and bias for the module
+    * @param newWeights
+    * @return true: set weights successfully
+    *         false: set weights failed
+    */
+  def setWeights(newWeights: Array[Tensor[T]]): Boolean = {
+    if (parameters() == null) {
+      false
+    } else {
+      val weights = parameters()._1
+      // set weight
```

Comment:
The first one won't always be a weight. parameters() may return (weight), (weight, bias) and (bias).
 Just add a require(weights.length == newWeights.length), then use a while loop to copy the weights.